### PR TITLE
[PHP 8.3] Assigning a negative index to an empty array

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Part of #2796

### CODE REVIEW

File: language/types/array.xml [hunk 1]
```diff
@@ -316,6 +316,50 @@ array(7) {
...
+}
+]]>
+    </screen>
+
+    <note>
+     <para>
+      Prior to PHP 8.3.0, assigning a negative integer key <literal>n</literal> would
+      assign the next key to <literal>0</literal>.
+     </para>
+    </note>
```
Comment (Girgias) [inline language/types/array.xml:361]: Could you also add a `screen` tag showcasing the old output?
```suggestion
    <warning>
     <simpara>
      Prior to PHP 8.3.0, assigning a negative integer key <literal>n</literal> would
      assign the next key to <literal>0</literal>.
     </simpara>
    </warning>
```

File: language/types/array.xml [hunk 2]
```diff
@@ -316,6 +316,50 @@ array(7) {
 ]]>
     </screen>
    </example>
+
+   <example>
+    <title>Negative index example</title>
+    <para>
+     When assigning a negative integer key <literal>n</literal>, PHP will take care to
+     assign the next key to <literal>n+1</literal>.
+    </para>
```
Comment (Girgias) [inline language/types/array.xml:325]:
```suggestion
    <simpara>
     When assigning a negative integer key <literal>n</literal>, PHP will take care to
     assign the next key to <literal>n+1</literal>.
    </simpara>
```

File: language/types/array.xml (hunk 1 above)
Comment (alexandre-daubois) [reply language/types/array.xml]: I also updated the example, it wasn't working as I expected in older versions. The new one is better: https://3v4l.org/J9ZRL

File: language/types/array.xml [hunk 3]
```diff
@@ -316,6 +316,56 @@ array(7) {
...
+  [-4]=>
+  int(2)
+}
+]]>
+    </screen>
+
+    <warning>
+     <simpara>
+      Prior to PHP 8.3.0, assigning a negative integer key <literal>n</literal> would
+      assign the next key to <literal>0</literal>, as shown in this example:
```
Comment (Girgias) [inline language/types/array.xml:353]: Maybe?
```suggestion
      Prior to PHP 8.3.0, assigning a negative integer key <literal>n</literal> would
      assign the next key to <literal>0</literal>, the previous example would therefore output:
```

Comment (alexandre-daubois) [reply language/types/array.xml]: Definitely!

**Pull request merged**